packages/daemon/src/types.d.ts

```diff
@@ -57,9 +57,35 @@ type FormulaIdentifierRecord = {
   number: string;
 };
 
+type EndoFormula = {
+  type: 'endo';
+  host: string;
+  leastAuthority: string;
+  webPageJs?: string;
```

Comment:
It would be neat to document what the `string` properties on all of our formulas are supposed to be. Names? Identifiers if the real thing? Identifiers of handles?

Reply:
how to make subtypes of string? so you can't give it a string unless you cast it to a formulaId? or not a good idea?

Reply:
Ah, I don't think it's worth doing anything with [template literal types](https://www.typescriptlang.org/docs/handbook/2/template-literal-types.html) here (especially in light of #2088). I just meant documenting what they're supposed to be on the interfaces would be nice.

I think we're OK for now since that information is there when the formulas are instantiated, so I won't block on it.